InventoryBuilder.py

```diff
@@ -243,92 +277,177 @@ def create_vcenter_objects(self, vrops, target: str, token: str):
+                          query_specs=query_specs.get('virtual_machines', {}))
+        distributed_virtual_switchs, self.response_codes[target]["distributed_virtual_switch"] = \
+            Vrops.get_dis_virtual_switch(vrops, target, token, [dc.uuid for dc in datacenter],
+                                         query_specs=query_specs.get('default', {}))
 
+        vcenter_adapter.datacenter = list()
         for dc in datacenter:
-            vcenter_adapter.add_datacenter(dc)
+            vcenter_adapter.datacenter.append(dc)
+            logger.debug(f'Collecting datacenter: {dc.name}')
+
         for dc_object in vcenter_adapter.datacenter:
-            logger.debug(f'Collecting datacenter: {dc_object.name}')
+            dc_object.datastores = list()
+            dc_object.clusters = list()
+            dc_object.dvss = list()
+
             for ds in datastores:
                 if ds.parent == dc_object.uuid:
-                    dc_object.add_datastore(ds)
+                    dc_object.datastores.append(ds)
                     logger.debug(f'Collecting datastore: {ds.name}')
             for cl in cluster:
-                dc_object.add_cluster(cl)
+                if cl.parent == dc_object.uuid:
+                    dc_object.clusters.append(cl)
+                    logger.debug(f'Collecting cluster: {cl.name}')
             for cl_object in dc_object.clusters:
-                logger.debug(f'Collecting cluster: {cl_object.name}')
+                cl_object.hosts = list()
                 for hs in hosts:
                     if hs.parent == cl_object.uuid:
-                        cl_object.add_host(hs)
+                        cl_object.hosts.append(hs)
+                        logger.debug(f'Collecting host: {hs.name}')
                 for hs_object in cl_object.hosts:
-                    logger.debug(f'Collecting host: {hs_object.name}')
+                    hs_object.vms = list()
                     for vm in vms:
                         if vm.parent == hs_object.uuid:
-                            hs_object.add_vm(vm)
+                            hs_object.vms.append(vm)
                             logger.debug(f'Collecting VM: {vm.name}')
+            for dvs in distributed_virtual_switchs:
+                if dvs.parent == dc_object.uuid:
+                    dc_object.dvss.append(dvs)
+                    logger.debug(f'Collecting distributed virtual switch: {dvs.name}')
         return vcenter_adapter
 
     def create_nsxt_objects(self, vrops, target: str, token: str):
-        nsxt_adapter, self.response_codes[target]["nsxt_adapter"] = Vrops.get_nsxt_adapter(vrops, target, token)
-        if not nsxt_adapter:
+        nsxt_adapter_list, self.response_codes[target]["nsxt_adapter"] = Vrops.get_nsxt_adapter(vrops, target, token)
+        if not nsxt_adapter_list:
+            logger.critical(f'Could not get any nsxt adapter!')
```

Comment:
Why using an f-string here? Maybe bring in some context which vrops we are looking at?